test/test_container.py

```diff
@@ -351,7 +331,7 @@ def raise_error():
         raise Exception('foobar')
 
     container.start()
-    container.spawn_managed_thread(raise_error)
+    container.spawn_managed_thread(raise_error, Extension())
```

Comment:
might change as discussed above, but if we're keeping it optional, could we not skip the warning if the value is `None`?